src/classes/Account.js

```diff
@@ -163,19 +163,23 @@ class Account {
 			.then(userID => api.get('stripe/customers', userID, 'cards,subscriptions'))
 			.then((res) => {
 				const customer = build(normalize(res), 'customers', res.data.id);
+
 				// TODO temporary fix to handle multiple subscriptions
 				let sub = customer.subscriptions;
 				if (!Array.isArray(sub)) {
 					sub = [sub];
 				}
-				const subPlus = sub.reduce((acc, curr) => {
-					let a = acc;
-					if (curr.productName.includes('Plus')) {
-						a = curr;
-					}
-					return a;
-				}, {});
-				this._setSubscriptionData(subPlus);
+
+				// Display premium info if user has both premium and plus subscriptions
+				const premiumSubscription = sub.find(subscription => subscription.productName.includes('Ghostery Premium'));
+				if (premiumSubscription) {
+					this._setSubscriptionData(premiumSubscription);
+				}
+
+				const plusSubscription = sub.find(subscription => subscription.productName.includes('Ghostery Plus'));
+				if (plusSubscription) {
+					this._setSubscriptionData(plusSubscription);
+				}
```

Comment:
Even though `account-web` call this function indirectly via the `accountPage.getuserSubscriptionData` message, it does not appear to actually use or even capture the return value. So the only (indirect, via the `account.getUserSubscriptionData` message handler in `background.js`) consumer of the return value is the GBE panel, via the `Account` component & reducer. So this is the only data flow out of the function that we need to be concerned with.

In this data flow, the message handler in `background.js` throws away most of the returned `customer` object, extracting the `Ghostery Premium` or `Ghostery Plus` entry from the `customer.subscriptions` array and sending just that back to the `Account` component in `app`, which is just what that component expects.

So, particularly since the extraction is duplicated here and in `background.js`, we can instead return just the `premiumSubscription`/`plusSubscription` object from this function directy, or else an empty object if neither subscription type is found.

Then, the handler in `background.js` can be simplified to

```
	if (name === 'account.getUserSubscriptionData') {
		account.getUserSubscriptionData()
			.then(subscriptionData => callback({ subscriptionData }))
			.catch((err) => {
				log('Error getting user subscription data:', err);
				callback({ errors: _getJSONAPIErrorsObject(err) });
			});
		return true;
	}
```

@christophertino  - would appreciate your input on whether this sounds like a safe change to make and also on whether you think we need to update what is being passed to the front end to reflect that the user could now have multiple subscriptions active simultaneously, or whether it's sufficient to pass the highest-tier subscription, allowing us to leave the `subscriptionData` Redux object and logic unchanged on the front end for now.

Reply:
Also, if we don't do this change, we need to make sure we don't clobber the Premium subscription set with the Plus subscription set in case a user somehow has both.

Reply:
Based on feedback from @christophertino , let's do a blend of what I suggested and what we have now. Instead of returning the whole `customer` object or just the highest tier sub, let's return an array with all the subscription objects (or an empty array, in case the user has neither sub type). So the function guarantees it returns an array. Then in the handler in `background`, we can extract the highest tier sub object from that array and return that, or else an empty object if the array is empty.

Reply:
Hey, thanks for digging into this and helping me understand why we can make this change. Love how you found a way to keep the return type of `account#getUserSubscriptionData` consistent as an array and allowed the respective handlers to be flexible

Reply:
Sure thing, this was fun to think about